### PR TITLE
updated the script to pick latest version

### PR DESCRIPTION
**Pull request by preetithakur**
script is updated so that it can pick the latest valgrind  version automatically
whenever its available.

Signed-off-by: preeti thakur <preeti.thakur@in.ibm.com>

[root@ltcever87-lp6 toolchain]# avocado run --test-runner runner valgrind.py
Fetching asset from valgrind.py:Valgrind.test
JOB ID     : a23ac84f87a1ee57703a9af355ab0b81dc54b4f6
JOB LOG    : /root/preeti/avocado-fvt-wrapper/results/job-2022-04-20T08.14-a23ac84/job.log
 (1/1) valgrind.py:Valgrind.test:  PASS (1966.78 s)
RESULTS    : PASS 1 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /root/preeti/avocado-fvt-wrapper/results/job-2022-04-20T08.14-a23ac84/results.html
JOB TIME   : 1997.65 s

### CODE REVIEW

File: toolchain/valgrind.py [hunk 1]
```diff
@@ -48,8 +48,9 @@ def setUp(self):
                 self.cancel('%s is needed for the test to be run' % package)
         run_type = self.params.get('type', default='upstream')
         if run_type == "upstream":
-            url = self.params.get('url', default="ftp://sourceware.org/pub/"
-                                  "valgrind/valgrind-3.18.1.tar.bz2")
+            url = "ftp://sourceware.org/pub/valgrind/"
+            version = self.params.get('version', default="3.18.1")
```
Comment (PraveenPenguin) [inline toolchain/valgrind.py:52]: please add this in ymal file

**Review by PraveenPenguin — CHANGES_REQUESTED** [review]

File: toolchain/valgrind.py [hunk 2]
```diff
@@ -48,8 +48,9 @@ def setUp(self):
...
+            url = '%s/valgrind-%s.tar.bz2' % (url, version)
```
Comment (Naresh-ibm) [inline toolchain/valgrind.py:53]: LGTM

**Review by Naresh-ibm — APPROVED** [review]

File: toolchain/valgrind.py (hunk 1 above)
Comment (Naresh-ibm) [reply toolchain/valgrind.py:52]: > please add this in yaml file
ya, I was above to quote this.

File: toolchain/valgrind.py.data/valgrind_source.yaml [hunk 3]
```diff
@@ -1,6 +1,7 @@
 run_type: !mux
     upstream:
-        url: 'ftp://sourceware.org/pub/valgrind/valgrind-3.18.1.tar.bz2'
+        url ="ftp://sourceware.org/pub/valgrind/"
+        version = self.params.get('version', default="3.18.1")
```
Comment (Naresh-ibm) [inline toolchain/valgrind.py.data/valgrind_source.yaml:4]: @preetithakur the right way to add parameter to the yaml is as below.
version: ""

I think url is not required here as it is defined in .py itself and doesn't need any input from yaml.
So only above line is enough but make sure the indentation part.

**Review by Naresh-ibm — CHANGES_REQUESTED** [review]

**Comment by Naresh-ibm** [comment]
LGTM

**Review by PraveenPenguin — APPROVED** [review]

**Pull request merged**